src/envoy/tcp/metadata_exchange/metadata_exchange.h

```diff
@@ -88,6 +86,16 @@ class MetadataExchangeConfig {
 
 using MetadataExchangeConfigSharedPtr = std::shared_ptr<MetadataExchangeConfig>;
 
+// A simple wrapper around generic values
+class MetadataExchangeState : public StreamInfo::FilterState::Object {
```

Comment:
I think you need to use WasmState for this to be accessible from within Wasm. The generic resolver is not aware of all filter states, but it knows how to unpack a WasmState.

Reply:
Thanks.. just created PR to expose WasmState: https://github.com/envoyproxy/envoy-wasm/pull/241